wiki/Client/Release_stream/Lazer/File_storage/en.md

```diff
@@ -0,0 +1,52 @@
+# File storage in osu!(lazer)
+
+By default, osu!(lazer) stores beatmaps, skins, and score replay files in the following directories:
+
+- `%appdata%/osu/files` (Windows),
+- `~/.local/share/osu/files` (Linux & macOS),
+- `Android/data/sh.ppy.osulazer/files/files` (Android).
+
+On desktop platforms, the file store can also be moved in its entirety to another location, using the `Change folder location...` button in client settings.
+
+## Storage structure
+
+osu!(stable)'s storage structure, where data was directly accessible to users as standard files, resulted in a lot of teething issues due to having to explicitly handle unwanted modifications of key files. To avoid this, osu!(lazer) employs a more stringent method of file storage. As such, there is no equivalent of the `Songs` and `Skins` folders. All files that are imported to lazer are stored under filenames that reflect their [SHA-256 hashes](https://en.wikipedia.org/wiki/SHA-2). Mappings to these files are held inside a client database.
+
+For example, a file with the SHA-256 hash of
+
+```
+1a47929b6056d34d25a95eeb2012395ceed66af6f40cc37c898a08482d6325d2
+```
+
+shall be stored under the path of
+
+```
+files/1/1a/1a47929b6056d34d25a95eeb2012395ceed66af6f40cc37c898a08482d6325d2
+```
+
+This allows for saving storage space by preventing duplicates of the same file from being persisted to disk, and prevents users (or other applications) from easily tampering with files that should not be tampered with. All inconveniences compared to stable that arise from this choice have been addressed by either allowing exports or adding new in-game management features, or will be addressed in due time by new features.
+
+At the time of writing, the correct procedure to manually modify skins or beatmaps outside of the facilities available in-game is to export the item, update it, and then reimport it again with the desired changes.
+
+## Migration from osu!(stable)
+
+### Via hard links
+
+On most Windows systems, osu!(lazer) will be able to import data from the stable version of the game without having to create a second copy of the data on disk. This is possible thanks to an operating system feature called *hard links*.
+
+A hard link is conceptually similar to a *shortcut* (also known as a *symbolic link*), in that it is a method that allows a user to have the same file available from multiple different places on their filesystem. However, while shortcuts are just plain files that point to a different file (and therefore require additional space), hard links work one level deeper, at the level of the filesystem itself.
```

Comment:
shortcuts and symbolic links on windows are a different beast, but might be fine to gloss over this for the user's sake here.

Reply:
I'll just snip the mention for simplicity.